## 1. Import the Pandas Library and Read the Dataset File in Spreadsheet Format based on the First - Fifth Debate Videos

### 1.1. DataFrame Variable from First Debate Video on Najwa Shihab YouTube Channel 

In [1]:
"""
Melakukan proses import library Pandas serta memanggil file dataset dari hasil
web scraping komentar video YouTube debat capres - cawapres babak pertama di channel YouTube Najwa Shihab
menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3 yang telah
disimpan atau di-export pada file berformat spreadsheet yang bernama "Export Comments Data from 1st Video Debate.xlsx"
"""

import pandas as pd

first_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 1st Video Debate.xlsx")

first_debate_video_df.head() # Menampilkan 5 data teratas yang dimiliki oleh variabel DataFrame "first_debate_video_df"

video_id                channel_id                  comment_id  \
0  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgxfMHgX25B2b3jkO4B4AaABAg   
1  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgzKC1PcDJZk7TvQ_Qh4AaABAg   
2  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugxtzzc3zKJ8nY3Sg7F4AaABAg   
3  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugypp7MrvLOhoz9GGUV4AaABAg   
4  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgyzdGCruWPoVqZjMul4AaABAg   

                author                                       text_comment  \
0       @zakishofi3442  Saya bukan pendukung siapapun, tapi terimakasi...   
1      @UmmuAzizah2010  Maa syaa Allaah Pak Anies ❤❤❤❤ <br>Terima kasi...   
2       @nursusilo7298  Tak terbantahkan Pak Anis cards yg paling siap...   
3      @abisopiyan8628  Saya dan keluarga berkat debat pertama ini ras...   
4  @chickennoodles1740  🇲🇨 <a href="https://www.youtube.com/watch?v=gU...   

   like_count  total_reply                                 channel_url  \
0        3182           91       http://www.youtube.com/@zakishofi3442   
1        1542           23      http://www.youtube.com/@UmmuAzizah2010   
2        1501           91       http://www.youtube.com/@nursusilo7298   
3        1452           54      http://www.youtube.com/@abisopiyan8628   
4        1130           49  http://www.youtube.com/@chickennoodles1740   

       published_date        updated_date  
0 2023-12-12 21:22:08 2023-12-12 21:22:08  
1 2023-12-12 16:27:52 2023-12-12 16:27:52  
2 2023-12-12 21:00:05 2023-12-12 21:00:05  
3 2023-12-12 17:49:04 2023-12-12 17:49:04  
4 2023-12-13 06:02:33 2023-12-13 06:02:33

In [2]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "first_debate_video_df"
first_debate_video_df.shape

(6500, 10)

In [3]:
# Menampilkan 28 data terbawah yang dimiliki oleh variabel DataFrame "first_debate_video_df"
first_debate_video_df.tail()

video_id                channel_id                  comment_id  \
6495  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgyjUMb7d6LvFBE6MLh4AaABAg   
6496  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgyTtGF-qdiyIO3tzuZ4AaABAg   
6497  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugwuh-9xsEErWG-t4x94AaABAg   
6498  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugw2SkGkDgmpmW8Vb7t4AaABAg   
6499  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugxbovw2xFcpo2G9Tgx4AaABAg   

               author                                       text_comment  \
6495          @idangz       pemilu sekarang bisa pks sama pdi akrab wkwk   
6496   @babadhani4569  kita kawal pemilu jangan ada kotak kotak kardu...   
6497       @ari_d_art  cawapres no2 g jadi di video call ya? positif ...   
6498  @ajosamiang2573  Kalau masalah hukum dan HAM menang pak Anies, ...   
6499   @bedsandycotto  Debat pertama ini,semua mengakui Pak Anis mena...   

      like_count  total_reply                             channel_url  \
6495           0            0          http://www.youtube.com/@idangz   
6496           0            0   http://www.youtube.com/@babadhani4569   
6497           0            0       http://www.youtube.com/@ari_d_art   
6498           0            0  http://www.youtube.com/@ajosamiang2573   
6499           0            0   http://www.youtube.com/@bedsandycotto   

          published_date        updated_date  
6495 2023-12-13 08:47:17 2023-12-13 08:47:17  
6496 2023-12-13 08:47:53 2023-12-13 08:49:41  
6497 2023-12-13 08:48:33 2023-12-13 08:49:12  
6498 2023-12-13 08:49:22 2023-12-13 08:49:22  
6499 2023-12-13 04:39:47 2023-12-13 04:39:47

### 1.2. DataFrame Variable from Second Debate Video on Najwa Shihab YouTube Channel 

In [4]:
"""
Melakukan proses pemanggilan (membaca) file dataset berformat Microsoft Excel (spreadsheet) dari hasil
web scraping komentar video YouTube debat capres - cawapres babak kedua di channel YouTube Najwa Shihab
menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

second_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 2nd Video Debate.xlsx")

second_debate_video_df.head(18) # Menampilkan 18 data teratas yang dimiliki oleh variabel DataFrame "second_debate_video_df"

video_id                channel_id                  comment_id  \
0   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgyJX6diRTVoG-ulwqB4AaABAg   
1   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  Ugxbn5S7DiuMkSkSbB94AaABAg   
2   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgzO_HcRVJSQ7ZrHCel4AaABAg   
3   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgwT9iJLafyZLuFcxTp4AaABAg   
4   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgzIiDvGzaNj1ANySFF4AaABAg   
5   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  Ugwbl-GWfFJp5LeN2Wt4AaABAg   
6   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  Ugwgp9OdMZbr4WzMMax4AaABAg   
7   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgwfHwGjcFGpSKPDJhJ4AaABAg   
8   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgxAI5v2FVOAEK5agid4AaABAg   
9   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgyfG8Ct3NCLKOeJoeV4AaABAg   
10  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgxMz3qyrtgGWYDViLt4AaABAg   
11  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgyBUjlqgxZFNI-9VsB4AaABAg   
12  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgwuazX3IVa7YtmVP754AaABAg   
13  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgyxbirYwitykwDnElR4AaABAg   
14  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  Ugw-9oOn7eMEADMk5M94AaABAg   
15  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgzPGUa_39zh19TfV_14AaABAg   
16  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  Ugwx23MnZp5NFdRO_gF4AaABAg   
17  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  Ugyca8M2CC8aZuJE0kV4AaABAg   

               author                                       text_comment  \
0      @luluk_layalie  Jujur, gw terpukau banget sama Mas Gibran. Ben...   
1   @erickvalvolt3622  awalnya ragu milih 02 krna wakilnya. akhirnya ...   
2   @ichadonnelly4634  Terbukti yang muda belum tentu tidak berkualit...   
3         @bibeat3672  Saya warga wonogiri bekerja di solo,saya meras...   
4           @Dn-cf3bz  Mantap. Indonesia sudah bergerak maju melebihi...   
5    @ansorisaleh3933  Setuju sama paslon no urut 2, harus lebih di s...   
6    @user-vt3ig7vv1u  Pasangan yg sangat cocok untuk negara ini, sal...   
7      @umarozaid6819  Waktunya anak muda mengguncang dunia menjadika...   
8        @aryad-lu8vs  Gibran ini tegas,berani,simple,ngak neko2,real...   
9   @yeni-erin-anesti  Pas debat capres, pa Prabowo bnr bnr di cecer ...   
10     @aingmacan3020  Jujur sy sebagai ibu yg punya anak msh kuliah ...   
11      @indopanca777  Lanjutkan gibran. Buktikan bahwa anda tidak ha...   
12   @muhamadmuna1007  Banyak yang meragukan wapres no 2. Pas disini ...   
13        @yunushasbi  dulu saya milih pak jokowi, dan sekarang saya ...   
14    @ongkidjami6351  Ini yang para buzzer bilang belimbing sayur, g...   
15        @suka_93LuT  Dan saya berganti memilih untuk no urut 2....<...   
16       @fcnasya7451  Siapapun yang kita dukung, semoga Allah jadika...   
17   @ruangreview9122  Paslon 2 memang cerdas memilih perwakilan  kom...   

    like_count  total_reply                               channel_url  \
0         1253           94     http://www.youtube.com/@luluk_layalie   
1         1133          100  http://www.youtube.com/@erickvalvolt3622   
2          989          108  http://www.youtube.com/@ichadonnelly4634   
3          798           46        http://www.youtube.com/@bibeat3672   
4          780           31          http://www.youtube.com/@Dn-cf3bz   
5          774           31   http://www.youtube.com/@ansorisaleh3933   
6          532           24   http://www.youtube.com/@user-vt3ig7vv1u   
7          525           17     http://www.youtube.com/@umarozaid6819   
8          520           34       http://www.youtube.com/@aryad-lu8vs   
9          498           50  http://www.youtube.com/@yeni-erin-anesti   
10         496           14     http://www.youtube.com/@aingmacan3020   
11         454           16      http://www.youtube.com/@indopanca777   
12         346            7   http://www.youtube.com/@muhamadmuna1007   
13         333           13        http://www.youtube.com/@yunushasbi   
14         323           42    http://www.youtube.com/@ongkidjami6351   
15    

In [5]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "second_debate_video_df"
second_debate_video_df.shape

(4500, 10)

### 1.3. DataFrame Variable from Third Debate Video on Najwa Shihab YouTube Channel

In [6]:
"""
Melakukan proses pemanggilan (membaca) file dataset berformat
Microsoft Excel (spreadsheet) dari hasil web scraping komentar video YouTube
debat capres - cawapres babak ketiga di channel YouTube Najwa Shihab
yang menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

third_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 3rd Debate Video.xlsx")

third_debate_video_df.head(7)  # Menampilkan 7 data teratas yang dimiliki oleh variabel DataFrame "third_debate_video_df"

video_id                channel_id                  comment_id  \
0  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  Ugx6mhUfDOUhd65fWyh4AaABAg   
1  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  Ugw1wHlLE405w38TGhx4AaABAg   
2  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  Ugwru57KZMbtoy1Y-Wp4AaABAg   
3  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  UgxrCbnzIEfsCua9aLZ4AaABAg   
4  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  Ugz511t5jjNh-dMbJux4AaABAg   
5  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  UgyCTTzakp21i32wylN4AaABAg   
6  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  UgxKZjD-QTaKFcBohId4AaABAg   

                     author  \
0          @udinlamidun9716   
1         @ameliaachtar1904   
2  @MuhammadDermawanSusanto   
3          @idincelebez3554   
4          @Sekarepe_deweke   
5            @lukiabizar478   
6              @bowshay2808   

                                        text_comment  like_count  total_reply  \
0  Siapa yg setuju 3 capres di undang bertiga di ...        3971          332   
1  Mudah2an pemilunya jujur, jadi sesuai pilihan ...        1396           54   
2  Pak Anies sadar bahwa Pak Prabowo secara teori...        1203          130   
3  Sangat setujuh sekali kalau mereka debat terbu...         985           32   
4  Jujur saja sejak debat awal sampai sampai deba...         556           47   
5  Ayo mata Najwa bedah semua isi kepala 3 capres...         550           17   
6  Siapa saja boleh bertanya , setiap pemimpin ha...         500           46   

                                       channel_url      published_date  \
0          http://www.youtube.com/@udinlamidun9716 2024-01-07 17:03:53   
1         http://www.youtube.com/@ameliaachtar1904 2024-01-07 16:47:46   
2  http://www.youtube.com/@MuhammadDermawanSusanto 2024-01-07 17:07:29   
3          http://www.youtube.com/@idincelebez3554 2024-01-07 21:49:21   
4          http://www.youtube.com/@Sekarepe_deweke 2024-01-07 17:18:30   
5            http://www.youtube.com/@lukiabizar478 2024-01-08 06:24:10   
6              http://www.youtube.com/@bowshay2808 2024-01-07 21:45:59   

         updated_date  
0 2024-01-09 23:18:49  
1 2024-01-07 16:47:46  
2 2024-01-07 17:07:29  
3 2024-01-07 21:49:21  
4 2024-01-07 17:18:30  
5 2024-01-08 06:24:10  
6 2024-01-07 21:45:59

In [7]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "third_debate_video_df"
third_debate_video_df.shape

(4500, 10)

### 1.4. DataFrame Variable from Fourth Debate Video on Najwa Shihab YouTube Channel

In [8]:
"""
Melakukan proses pemanggilan/membaca file dataset yang disimpan pada Microsoft Excel (spreadsheet) dari hasil
web scraping komentar video YouTube debat capres - cawapres babak keempat di channel YouTube Najwa Shihab
yang menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

fourth_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 4th Video Debate.xlsx")

fourth_debate_video_df.head(24) # Menampilkan 24 data teratas yang dimiliki oleh variabel DataFrame "fourth_debate_video_df"

video_id                channel_id                  comment_id  \
0   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwTpSK1hjkTNQlcT894AaABAg   
1   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgzaWyuIp8PPLQrUOct4AaABAg   
2   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwRwfdnlejN5_d0iCp4AaABAg   
3   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  Ugwh7t7Q0xR4zuXgTMd4AaABAg   
4   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  Ugy4OpeQ0Eha8zicbmd4AaABAg   
5   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwVh8rFo3I5AXjpA1p4AaABAg   
6   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgyOVFSw_uMl6wnZ0e14AaABAg   
7   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwYLWQJqM8cRj8MAXx4AaABAg   
8   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgyCbia7NddL0pVJicR4AaABAg   
9   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  Ugx9HVSGU9bydZgEbf14AaABAg   
10  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  Ugy9V4RAS-TlVOwz6AR4AaABAg   
11  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgyV6sk5uUOzZRoV1c94AaABAg   
12  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwcsGsc4wo-ke8mnJ14AaABAg   
13  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  Ugxi_sR33CQMm7gLF0t4AaABAg   
14  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgyQx6qqIveEN2PMltF4AaABAg   
15  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  Ugw6orSayaixBOb8kKh4AaABAg   
16  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgxQENsGJ5SNIsq2Gkp4AaABAg   
17  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgyEpPsLAo87icVCfYN4AaABAg   
18  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  Ugy3eoVARblaiJ1LOVJ4AaABAg   
19  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwWSUYqmOXP8Bf1-dJ4AaABAg   
20  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwZD6LiDJBC8K5pLqB4AaABAg   
21  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgzBaXhHwaZgtzKOMol4AaABAg   
22  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgyfCasBBUSzzZ-jlWp4AaABAg   
23  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgyDrjbmHaMu7vcveJ14AaABAg   

                           author  \
0                  @sandigustiana   
1               @abizardfahry4591   
2                 @RickyFahrezi13   
3            @achmadzaenuddin9571   
4              @msgteamproduction   
5              @misbahulmunir1246   
6               @ningsihsiska1537   
7   @dinayuspitasarissiaptmsi3379   
8                   @ambiosaputra   
9                    @adnan2ganni   
10                     @penaku851   
11               @teukuhafizh5732   
12              @diditsaputra4138   
13                @zazanyunyu1814   
14        @semeleketeofficial9813   
15                 @rizalfikri525   
16               @user-wp3zh9rv9r   
17                     @nasralhaq   
18                      @KataSeRi   
19                  @jamilawilda_   
20            @ikrievanuraini7111   
21                 @akbarsany5609   
22                  @Andi_Fadilah   
23                @electrolux9808   

                                         text_comment  like_count  \
0   Timestamp<br><br><a href="https://www.youtube....        2936   
1   Siapa yang disini setuju 3 capres dan cawapres...        2532   
2   Saya pendukung AMIN.<br>Saya salut kpd PAK mah...        1111   
3   Terharu lihat pak Anies dan Gus Imin sangat me...         689   
4   Tetaplah jadi &quot;narasi&quot; untuk semua. ...         606   
5   Gue orang madura dan dari kemarin lebih suka l...         537   
6   Gilak memang, lebih seru nonton disini dari pa...         502   
7   sekelas Prof, klo mau mengeluarkan vocab yg ac...         481   
8   Liat sepak terjang Gibran jadi cawapres, makin...         435   
9         Cuma Abdur yang bukan mantan narapidana 😮😮😮         432   
10  Gw pendukung 01 tapi suka banget penjelasan da...         351   
11  <a href="https://www.youtube.com/watch?v=tEbRB...         323   
12  Semoga Abdur sehat selalu untuk mendukung 01 AMIN         318   
13  Salut sama pak.mahfud malah dia kasih wejangan...         301   
14        Semakin mantap tidak memilih 02 asam sulfat         294   
15  Melihat pejabat aktif mendukung salah satu pas...         290   
16  Kami orang madura bangga mempunyai prof.mahfud...         259   
17  Dialog An

In [9]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "fourth_debate_video_df"
fourth_debate_video_df.shape

(3999, 10)

### 1.5. DataFrame Variable from Fifth Debate Video on Najwa Shihab YouTube Channel

In [10]:
"""
Melakukan proses pemanggilan/membaca file dataset yang disimpan pada Microsoft Excel (spreadsheet) dari hasil
web scraping komentar video YouTube debat capres - cawapres babak kelima di channel YouTube Najwa Shihab
yang menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

fifth_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 5th Video Debate.xlsx")

fifth_debate_video_df.head(33) # Menampilkan 33 data teratas yang dimiliki oleh variabel DataFrame "fifth_debate_video_df"

video_id                channel_id                  comment_id  \
0   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgwB34PptWoq1w3z7xN4AaABAg   
1   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgxcIoHuK_A7ifquMpJ4AaABAg   
2   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzrVJ1HRPmYF26uAOx4AaABAg   
3   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgyKD5bEWEUpVcmbA7Z4AaABAg   
4   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgxBPscccnmnAWn4u-94AaABAg   
5   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugxp0PDYBoecwfODRCd4AaABAg   
6   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugzlw6mcCU3WbcCKDzF4AaABAg   
7   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzHGuNpDx56spVFDLh4AaABAg   
8   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzuTnI2lpmzcijB2Ap4AaABAg   
9   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugxdj1yoe3hL8z3Fpw14AaABAg   
10  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgwUTblfqAJX95dvfad4AaABAg   
11  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzzMXw-YF_IvdX_Ish4AaABAg   
12  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugzy89HQxStdI-SB5Wt4AaABAg   
13  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzcjpXnbTws9AKP38V4AaABAg   
14  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugx1ENjYxybL67XmwR54AaABAg   
15  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzeESCB_eiFbCHiAqh4AaABAg   
16  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugx5r2XFbrspyDRtcCV4AaABAg   
17  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgwyGwzU1CPbxV4vKjZ4AaABAg   
18  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugxvp-vpirUUKXchW-B4AaABAg   
19  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzA0L4av4yYxA1s25J4AaABAg   
20  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugw4jSAGci2o7nmd8bF4AaABAg   
21  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzfCWwlzQ8QcPdF3OV4AaABAg   
22  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugwf21BwWJw-RMS9uQt4AaABAg   
23  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugyw9X044-whyABVAkJ4AaABAg   
24  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgxF-pwUei2xtnyhMq54AaABAg   
25  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugwp0f6FNAhJCXWzdeF4AaABAg   
26  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugwfkl0LOHG27wKutYF4AaABAg   
27  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzH2JY0VFzYfwMuxrV4AaABAg   
28  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzgC4-Ga5J1QVfwu6t4AaABAg   
29  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgyeIMJAQ1B3bavd9c14AaABAg   
30  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugy_jbvkN5T6LvqviZN4AaABAg   
31  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgyCjlOG7Lujdaf1hct4AaABAg   
32  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugxz2eB4yWEbvgghDol4AaABAg   

                            author  \
0         @lilissuryanilimbong7685   
1                 @Millenial_IQ181   
2                  @edysupaidi8586   
3                     @aabasith622   
4                 @user-nv5dd1wg6l   
5                   @sandigustiana   
6                        @kancut26   
7                 @ludimaulana8304   
8                          @SiBORR   
9                      @faizadie30   
10                 @ihsanteras4940   
11                 @mochtarrajak96   
12                    @azuki_gxcha   
13                   @mandaoji9958   
14                    @apibara8712   
15                @user-kz5ze9nd8d   
16                        @WWFTV19   
17              @forcemitrasulthan   
18           @enisititurmiasih8003   
19                 @eviedavina3043   
20                     @Gen.zieeee   
21                @user-sl4cn8ez2g   
22  @elfm.englishlessonformusli173   
23                    @donrizal613   
24                 @ryanharian5191   
25                   @aldivatv5513   
26           @muhammadarahmaan0806   
27                       @aldy7740   
28                  @Anisalfatih92   
29                     @uyeere1810   
30                @olihsusanto7173   
31           @ramandaputriyani5592   
32                 @dariseliza5736   

                                         text_comment  like_count  \
0   Selaku Ibu Rumah Tangga saya tahu apa yg dibut...        1487   
1   Terimakasih Bapak H. Anies Rasyid Baswedan, S....         940   
2   Aplous buat Paslon 0

In [11]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "fifth_debate_video_df"
fifth_debate_video_df.shape

(3500, 10)

In [12]:
# Menampilkan 6 data terbawah yang dimiliki oleh variabel DataFrame "fifth_debate_video_df"
fifth_debate_video_df.tail(6)

video_id                channel_id                  comment_id  \
3494  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgxxcL0nr-Xcs6e9u4N4AaABAg   
3495  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugy6UhepaIM4GFGkool4AaABAg   
3496  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgwklhXMYbZ_RSj-NiB4AaABAg   
3497  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzDNnXWJ9VtebCN_Kt4AaABAg   
3498  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugwk1CGnKC0Nz98LV694AaABAg   
3499  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzfXPT4ZqEF_z68Qox4AaABAg   

                         author  \
3494          @adeliasandra1894   
3495       @stephentambanon6735   
3496              @imanhaji9510   
3497              @BeranjakGede   
3498  @adhilistyohadribroto8306   
3499           @davidrianto7895   

                                           text_comment  like_count  \
3494                         Audio mba nana ko ilang ya           0   
3495  Gk ada suaranya  mulai <a href="https://www.yo...           0   
3496                          Ini gak ada suaranya yaa?           0   
3497  <a href="https://www.youtube.com/watch?v=JFl2f...           0   
3498  mbah wo suka jump conclussion tidak mencari ak...           0   
3499                                   Audio nya bos...           0   

      total_reply                                       channel_url  \
3494            0          http://www.youtube.com/@adeliasandra1894   
3495            0       http://www.youtube.com/@stephentambanon6735   
3496            0              http://www.youtube.com/@imanhaji9510   
3497            2              http://www.youtube.com/@BeranjakGede   
3498            0  http://www.youtube.com/@adhilistyohadribroto8306   
3499            0           http://www.youtube.com/@davidrianto7895   

          published_date        updated_date  
3494 2024-02-05 01:51:04 2024-02-05 01:51:04  
3495 2024-02-05 01:50:57 2024-02-05 01:50:57  
3496 2024-02-05 01:50:08 2024-02-05 01:50:08  
3497 2024-02-05 01:50:03 2024-02-05 01:50:03  
3498 2024-02-05 01:49:01 2024-02-05 01:49:01  
3499 2024-02-05 01:01:35 2024-02-05 01:01:35

## 2. Combining Five DataFrame Variables Contain Comments Information from the First - Fifth Debate Videos on Najwa Shihab YouTube Channel

In [13]:
"""
Menggabungkan 5 variabel DataFrame menjadi 1 file dataset dalam format
spreadsheet Microsoft Excel dimana masing-masing variabel DataFrame memiliki jumlah
data yang disimpan berbeda-beda karena saat jumlah data yang didefinisikan saat
melakukan proses scraping/pengambilan informasi-informasi komentar utama dengan metode pengulangan (looping)
dibuat berbeda antara jumlah data komentar dari video debat pertama, debat kedua, dst
"""

df_comment_video_debate_najwashihab = pd.concat([first_debate_video_df, second_debate_video_df, third_debate_video_df,
                                                 fourth_debate_video_df, fifth_debate_video_df], ignore_index = True)

# Menampilkan 5 data teratas yang dimiliki oleh variabel DataFrame "df_comment_video_debate_najwashihab"
df_comment_video_debate_najwashihab.head()

video_id                channel_id                  comment_id  \
0  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgxfMHgX25B2b3jkO4B4AaABAg   
1  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgzKC1PcDJZk7TvQ_Qh4AaABAg   
2  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugxtzzc3zKJ8nY3Sg7F4AaABAg   
3  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugypp7MrvLOhoz9GGUV4AaABAg   
4  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgyzdGCruWPoVqZjMul4AaABAg   

                author                                       text_comment  \
0       @zakishofi3442  Saya bukan pendukung siapapun, tapi terimakasi...   
1      @UmmuAzizah2010  Maa syaa Allaah Pak Anies ❤❤❤❤ <br>Terima kasi...   
2       @nursusilo7298  Tak terbantahkan Pak Anis cards yg paling siap...   
3      @abisopiyan8628  Saya dan keluarga berkat debat pertama ini ras...   
4  @chickennoodles1740  🇲🇨 <a href="https://www.youtube.com/watch?v=gU...   

   like_count  total_reply                                 channel_url  \
0        3182           91       http://www.youtube.com/@zakishofi3442   
1        1542           23      http://www.youtube.com/@UmmuAzizah2010   
2        1501           91       http://www.youtube.com/@nursusilo7298   
3        1452           54      http://www.youtube.com/@abisopiyan8628   
4        1130           49  http://www.youtube.com/@chickennoodles1740   

       published_date        updated_date  
0 2023-12-12 21:22:08 2023-12-12 21:22:08  
1 2023-12-12 16:27:52 2023-12-12 16:27:52  
2 2023-12-12 21:00:05 2023-12-12 21:00:05  
3 2023-12-12 17:49:04 2023-12-12 17:49:04  
4 2023-12-13 06:02:33 2023-12-13 06:02:33

Berdasarkan code diatas, saya telah berhasil menggabungkan 5 variabel DataFrame yang masing-masing DataFrame memiliki data/informasi komentarnya masing-masing. Sebagai contoh pada variabel DataFrame "first_debate_video_df" berisikan 6.500 data/informasi komentar yang berhasil diambil dari video debat babak pertama capres-cawapres Republik Indonesia tahun 2024 yang ditayangkan oleh YouTube channel Najwa Shihab. Sedangkan variabel DataFrame "second_debate_video_df" juga memiliki data/informasi komentar utama namun yang membedakannya dengan DataFrame "first_debate_video_df" adalah jumlah data yang diambil dan sumber videonya karena jika variabel DataFrame "second_debate_video_df" memiliki 4.500 dan sumber informasi/data komentar yang diambil berasal dari video debat babak kedua capres-cawapres Republik Indonesia tahun 2024 yang di-upload oleh YouTube channel Najwa Shihab.

Hal tersebut juga sama dengan yang dimiliki oleh variabel DataFrame lainnya seperti "third_debate_video_debate_df", "fourth_debate_video_df" dan "fifth_debate_video_df" dengan catatan jumlah data yang diambil berbeda-beda dan sumber video atas informasi komentar yang diambil juga berbeda namun video-video tersebut tetap di-upload/ditayangkan oleh YouTube channel Najwa Shihab.

Untuk mengecek dan memastikan apakah penggabungan 5 variabel DataFrame berhasil dilakukan, saya akan menampilkan 6 data terbawah yang dimiliki oleh DataFrame "df_comment_video_debate_najwashihab" karena jika proses penggabungan 5 variabel DataFrame menjadi 1 DataFrame berhasil dilakukan maka 6 data terbawah seharusnya berasal dari data-data komentar yang diambil (di-scraping) dari video debat babak kelima capres-cawapres Republik Indonesia tahun 2024 yang ditayangkan oleh YouTube channel Najwa Shihab dan disimpan di dalam variabel DataFrame bernama "fifth_debate_video_df".

Sebagai penjelasan tambahan saat saya melakukan proses penggabungan 5 variabel DataFrame yang berbeda-beda menjadi 1 DataFrame agar data-data komentar dari video debat babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube Najwa Shihab tidak terpisah-pisah dan tersentralisasi (terpusat) ke dalam 1 file dataset, saya menggunakan sebuah parameter yaitu ignore_index yang bernilai Boolean yaitu True. Parameter tersebut saya gunakan (ignore_index = True) bertujuan agar indeks pada DataFrame (dalam hal ini indeks yang terdapat di dalam variabel DataFrame "df_comment_debate_video_najwashihab") tidak terduplikasi dan tidak mengulangi nilai indeks yang sama dari indeks awal.

Sebagai contoh variabel DataFrame "first_video_debate_df" memiliki sebuah nilai awal indeks yaitu 0 (indeks data 0 mengartikan jika data tersebut merupakan baris pertama yang ada di dalam data tersebut) dan indeks terakhir pada variabel DataFrame "first_video_debate_df" adalah 6499 yang mengartikan bahwa baris data tersebut merupakan baris data ke-6.500. Lalu karena DataFrame "first_video_debate_df" digabungkan dengan DataFrame "second_video_debate_df" maka baris data selanjutnya sudah pasti merupakan data yang dimiliki oleh indeks data ke-0/baris pertama dari DataFrame "second_video_debate_df".

Jika tidak menggunakan parameter ignore_index yang bernilai True, maka variabel DataFrame "df_comment_video_debate_najwashihab" yang telah menjadi DataFrame tersentralisasi tidak melanjutkan nilai indeks yang selanjutnya yaitu indeks data ke-6500 yang merupakan data pertama/indeks data ke-0 dari variabel DataFrame "second_video_debate_df". Jadi secara tidak langsung akan ada 2 baris data yang memiliki indeks ke-0 jika DataFrame "df_comment_video_debate_najwashihab" berisikan data-data komentar dari DataFrame "first_video_debate_df" dan "second_video_debate_df". Karena DataFrame yang tersentralisasi ini berisikan data-data komentar hasil penggabungan 5 variabel DataFrame yang berbeda, maka jika tidak menggunakan parameter ignore_index bernilai True DataFrame "df_comment_debate_video_najwashihab" memiliki 5 baris data yang memiliki indeks data ke-0

Jadi alasan tersebutlah saya menggunakan parameter ignore_index = True untuk menghindari terjadinya duplikasi indeks data pada DataFrame yang tersentralisasi/terpusat yaitu variabel DataFrame "df_comment_debate_video_najwashihab".

In [14]:
# Menampilkan 6 data terbawah yang dimiliki oleh variabel DataFrame "df_comment_video_debate_najwashihab"
df_comment_video_debate_najwashihab.tail(6)

video_id                channel_id                  comment_id  \
22993  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgxxcL0nr-Xcs6e9u4N4AaABAg   
22994  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugy6UhepaIM4GFGkool4AaABAg   
22995  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgwklhXMYbZ_RSj-NiB4AaABAg   
22996  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzDNnXWJ9VtebCN_Kt4AaABAg   
22997  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  Ugwk1CGnKC0Nz98LV694AaABAg   
22998  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzfXPT4ZqEF_z68Qox4AaABAg   

                          author  \
22993          @adeliasandra1894   
22994       @stephentambanon6735   
22995              @imanhaji9510   
22996              @BeranjakGede   
22997  @adhilistyohadribroto8306   
22998           @davidrianto7895   

                                            text_comment  like_count  \
22993                         Audio mba nana ko ilang ya           0   
22994  Gk ada suaranya  mulai <a href="https://www.yo...           0   
22995                          Ini gak ada suaranya yaa?           0   
22996  <a href="https://www.youtube.com/watch?v=JFl2f...           0   
22997  mbah wo suka jump conclussion tidak mencari ak...           0   
22998                                   Audio nya bos...           0   

       total_reply                                       channel_url  \
22993            0          http://www.youtube.com/@adeliasandra1894   
22994            0       http://www.youtube.com/@stephentambanon6735   
22995            0              http://www.youtube.com/@imanhaji9510   
22996            2              http://www.youtube.com/@BeranjakGede   
22997            0  http://www.youtube.com/@adhilistyohadribroto8306   
22998            0           http://www.youtube.com/@davidrianto7895   

           published_date        updated_date  
22993 2024-02-05 01:51:04 2024-02-05 01:51:04  
22994 2024-02-05 01:50:57 2024-02-05 01:50:57  
22995 2024-02-05 01:50:08 2024-02-05 01:50:08  
22996 2024-02-05 01:50:03 2024-02-05 01:50:03  
22997 2024-02-05 01:49:01 2024-02-05 01:49:01  
22998 2024-02-05 01:01:35 2024-02-05 01:01:35

Berdasarkan hasil diatas dapat diketahui jika indeks data ke-22998 atau yang merupakan baris data 22.999 sama dengan data komentar yang dimiliki oleh variabel DataFrame "fifth_video_debate_df" yang merupakan indeks data ke-3499 atau pada baris data 3.500 sehingga proses penggabungan data-data komentar yang didapatkan dari video debat babak pertama hingga babak kelima capres-cawapres Republik Indonesia tahun 2024 telah berhasil dilakukan dan urutan indeks dari variabel DataFrame yang tersentralisasi (df_comment_video_debate_najwashihab) sudah sesuai dengan output yang diharapkan.

In [15]:
# Menampilkan jumlah data duplikat yang dimiliki di dalam variabel DataFrame "df_comment_debate_video_najwashihab"
df_comment_video_debate_najwashihab.duplicated().sum()

0

In [16]:
"""
Menampilkan jumlah nilai data yang hilang (missing values) dari masing-masing
kolom yang terdapat di dalam variabel DataFrame "df_comment_video_debate_najwashihab"
"""
df_comment_video_debate_najwashihab.isnull().sum()

video_id          0
channel_id        0
comment_id        0
author            0
text_comment      0
like_count        0
total_reply       0
channel_url       0
published_date    0
updated_date      0
dtype: int64

In [17]:
"""
Menampilkan beberapa informasi dasar dari setiap kolom seperti tipe data,
jumlah baris data atau kolom yang dimiliki oleh DataFrame, dll., yang dimiliki
oleh variabel DataFrame "df_comment_video_debate_najwashihab"
"""
df_comment_video_debate_najwashihab.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22999 entries, 0 to 22998
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   video_id        22999 non-null  object        
 1   channel_id      22999 non-null  object        
 2   comment_id      22999 non-null  object        
 3   author          22999 non-null  object        
 4   text_comment    22999 non-null  object        
 5   like_count      22999 non-null  int64         
 6   total_reply     22999 non-null  int64         
 7   channel_url     22999 non-null  object        
 8   published_date  22999 non-null  datetime64[ns]
 9   updated_date    22999 non-null  datetime64[ns]
dtypes: datetime64[ns](2), int64(2), object(6)
memory usage: 1.8+ MB


Berdasarkan output yang didapatkan dari fungsi "info()" yang menunjukkan beberapa informasi dasar yang terdapat di dalam DataFrame, kita dapat mengetahui beberapa poin informasi dari variabel DataFrame "df_comment_video_debate_najwashihab" diantaranya adalah sebagai berikut:

1. Variabel DataFrame "df_comment_video_debate_najwashihab" merupakan sebuah objek berupa DataFrame yang berisikan kumpulan data yang didalamnya terdapat baris, kolomda dan tabel (dataset) dari hasil proses penggabungan beberapa data-data komentar yang didapatkan dari video YouTube tentang debat calon Presiden-Wakil Presiden Republik Indonesia tahun 2024 babak pertama hingga  babak kelima yang ditayangkan oleh channel YouTube Najwa Shihab.

2. Variabel DataFrame "df_comment_video_debate_najwashihab" memiliki jumlah data/baris sebanyak 22.999 data yang terdiri dari teks komentar yang dikirimkan oleh penonton video tersebut, nama akun yang berkomentar, tanggal dan waktu dari komentar tersebut dibuat, dll. Data-data tersebut didapatkan melalui proses penggabungan kelima variabel DataFrame yang masing-masing memiliki jumlah data/informasi komentar yang berbeda-beda seperti pada variabel DataFrame "first_debate_video_df" memiliki baris data sebanyak 6.500 informasi komentar, variabel DataFrame "second_debate_video_df" memiliki baris data sebanyak 4.500 data komentar, dst.

3. DataFrame "df_comment_video_debate_najwashihab" memiliki 10 kolom yang diantaranya adalah kolom "video_id", "channel_id", "comment_id", "author", "text_comment", "like_count", "total_reply", "channel_url", "published_date" dan terakhir adalah kolom "updated_date".

4. Masing-masing kolom yang ada di dalam DataFrame "df_comment_video_debate_najwashihab" tidak memiliki nilai NaN (Not a Number) atau nilai yang hilang (missing value) sehingga dapat disimpulkan bahwa sepuluh kolom yang dimiliki oleh variabel DataFrame "df_comment_video_debate_najwashihab" dan setiap baris indeksnya dimiliki oleh 10 kolom, tidak ada satupun baris data pada indeks yang tidak memiliki nilai/data.

5. Pada variabel DataFrame "df_comment_video_debate_najwashihab" terdapat 3 jenis tipe data yang berbeda. Tipe data yang pertama adalah tipe data Integer untuk setiap kolom yang memiliki nilai data bersifat yang angka atau numerik. Data numerik/data angka dibedakan menjadi 2 jenis bilangan angka yaitu bilangan bulat dan bilangan desimal. Angka bilangan bulat direpresentasikan sebagai tipe data Integer dan angka bilangan desimal direpresentasikan sebagai tipe data Float. Karena pada DataFrame "df_comment_video_debate_najwashihab" hanya memiliki 2 kolom saja yang bertipe data Integer, yaitu kolom "like_count" dan "total_reply", di mana kedua kolom tersebut memiliki nilai angka bilangan bulat, maka kolom tersebut bertipe data Integer bukan Float. Tipe data yang kedua adalah tipe data Datetime dan hanya 2 kolom saja yang memiliki tipe data Datetime yaitu kolom "published_date" dan "updated_date". Tipe data yang ketiga yang dimiliki oleh variabel DataFrame "df_comment_video_debate_najwashihab" adalah tipe data Object/String. Tipe data Object/String adalah representasi dari nilai-nilai data teks, simbol-simbol maupun karakter spesial yang dimiliki oleh suatu kolom di dalam DataFrame. Secara jumlah, tipe data Object/String pada variabel DataFrame "df_comment_video_debate_najwashihab" dimiliki oleh 6 kolom (terbanyak diantara tipe data Integer ataupun Datetime) yaitu kolom "video_id", "channel_id", "comment_id", "author", "text_comment" dan kolom "channel_url".

In [18]:
"""
Mengubah urutan data yang dimiliki oleh variabel DataFrame "df_comment_video_debate_najwashihab"
berdasarkan jumlah like terbanyak hingga jumlah like paling sedikit yang didapatkan/diterima oleh suatu teks komentar
diantara 22.999 data komentar yang merupakan kumpulan data hasil penggabungan komentar-komentar dari video
debat babak pertama hingga babak kelima capres-cawapres Republik Indonesia yang ditayangkan oleh channel YouTube Najwa Shihab
"""

df_comment_video_debate_najwashihab = df_comment_video_debate_najwashihab.sort_values(by=["like_count"],
                                                                                      ascending = False, ignore_index = True)

df_comment_video_debate_najwashihab.head(8) # Menampilkan 8 data komentar teratas yang memiliki jumlah like terbanyak

video_id                channel_id                  comment_id  \
0  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  Ugx6mhUfDOUhd65fWyh4AaABAg   
1  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgxfMHgX25B2b3jkO4B4AaABAg   
2  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwTpSK1hjkTNQlcT894AaABAg   
3  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgzaWyuIp8PPLQrUOct4AaABAg   
4  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgzKC1PcDJZk7TvQ_Qh4AaABAg   
5  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugxtzzc3zKJ8nY3Sg7F4AaABAg   
6  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgwB34PptWoq1w3z7xN4AaABAg   
7  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugypp7MrvLOhoz9GGUV4AaABAg   

                     author  \
0          @udinlamidun9716   
1            @zakishofi3442   
2            @sandigustiana   
3         @abizardfahry4591   
4           @UmmuAzizah2010   
5            @nursusilo7298   
6  @lilissuryanilimbong7685   
7           @abisopiyan8628   

                                        text_comment  like_count  total_reply  \
0  Siapa yg setuju 3 capres di undang bertiga di ...        3971          332   
1  Saya bukan pendukung siapapun, tapi terimakasi...        3182           91   
2  Timestamp<br><br><a href="https://www.youtube....        2936          124   
3  Siapa yang disini setuju 3 capres dan cawapres...        2532           91   
4  Maa syaa Allaah Pak Anies ❤❤❤❤ <br>Terima kasi...        1542           23   
5  Tak terbantahkan Pak Anis cards yg paling siap...        1501           91   
6  Selaku Ibu Rumah Tangga saya tahu apa yg dibut...        1487          169   
7  Saya dan keluarga berkat debat pertama ini ras...        1452           54   

                                       channel_url      published_date  \
0          http://www.youtube.com/@udinlamidun9716 2024-01-07 17:03:53   
1            http://www.youtube.com/@zakishofi3442 2023-12-12 21:22:08   
2            http://www.youtube.com/@sandigustiana 2024-01-21 21:32:13   
3         http://www.youtube.com/@abizardfahry4591 2024-01-22 00:17:18   
4           http://www.youtube.com/@UmmuAzizah2010 2023-12-12 16:27:52   
5            http://www.youtube.com/@nursusilo7298 2023-12-12 21:00:05   
6  http://www.youtube.com/@lilissuryanilimbong7685 2024-02-04 20:29:41   
7           http://www.youtube.com/@abisopiyan8628 2023-12-12 17:49:04   

         updated_date  
0 2024-01-09 23:18:49  
1 2023-12-12 21:22:08  
2 2024-01-21 21:32:13  
3 2024-01-22 00:17:18  
4 2023-12-12 16:27:52  
5 2023-12-12 21:00:05  
6 2024-02-04 20:29:41  
7 2023-12-12 17:49:04

## 3. Export DataFrame Variable into Dataset with Spreadsheet Format (Microsoft Excel)

In [19]:
"""
Melakukan proses penyimpanan/meng-eksport data atau informasi komentar utama sebanyak 22.999 dari hasil
penggabungan data-data dari kelima variabel DataFrame yang didapatkan dari hasil web scraping pada
video debat capres-cawapres babak pertama hingga babak kelima yang di-upload oleh Channel YouTube Najwa Shihab
"""

df_comment_video_debate_najwashihab.to_excel("../Dataset/Dataset Komentar Video Debat dari YouTube Channel Najwa Shihab.xlsx",
                                             index = False, header = True)